src/tribler-common/tribler_common/sentry_reporter/sentry_reporter.py

```diff
@@ -0,0 +1,319 @@
+    _logger = logging.getLogger('SentryReporter')
+
+    @staticmethod
+    def init(sentry_url='', scrubber=None):
+        """ Initialization.
+
+        This method should be called in each thread that uses SentryReporter.
+
+        Args:
+            sentry_url: URL for Sentry server. If it is empty then Sentry's
+                sending mechanism will not be initialized.
+
+            scrubber: a class that will be used for scrubbing sending events.
+                Only a single method should be implemented in the class:
+                ```
+                    def scrub_event(self, event):
+                        pass
+                ```
+        Returns:
+            Sentry Guard.
+        """
+        SentryReporter._logger.info(f"Init: {sentry_url}")
+
+        SentryReporter._scrubber = scrubber
+        return sentry_sdk.init(
+            sentry_url,
+            release=None,
+            # https://docs.sentry.io/platforms/python/configuration/integrations/
+            integrations=[
+                LoggingIntegration(
+                    level=logging.INFO,  # Capture info and above as breadcrumbs
+                    event_level=logging.ERROR  # Send errors as events
+                ),
+            ],
+            before_send=SentryReporter._before_send,
+        )
+
+    @staticmethod
+    def send(event, post_data, sys_info):
+        """Send the event to the Sentry server
+
+        This method
+            1. Enable Sentry's sending mechanism.
+            2. Extend sending event by the information from post_data.
+            3. Send the event.
+            4. Disables Sentry's sending mechanism.
+
+        Scrubbing the information will be performed in the `_before_send` method.
+
+        During the execution of this method, all unhandled exceptions that
+        will be raised, will be sent to Sentry automatically.
+
+        Args:
+            event: event to send. It should be taken from SentryReporter at
+            post_data: dictionary made by the feedbackdialog.py
+                previous stages of executing.
+            sys_info: dictionary made by the feedbackdialog.py
+
+        Returns:
+            Event that was sent to Sentry server
+        """
+        SentryReporter._logger.info(f"Send: {post_data}, {event}")
+
+        if event is None:
+            return event
+
+        with AllowSentryReports(value=True, description='SentryReporter.send()'):
+            # prepare event
+            if CONTEXTS not in event:
+                event[CONTEXTS] = {}
+
+            if TAGS not in event:
+                event[TAGS] = {}
+
+            event[CONTEXTS][REPORTER] = {}
+
+            # tags
+            tags = event[TAGS]
+            tags['version'] = safe_get(post_data, 'version', None)
+            tags['machine'] = safe_get(post_data, 'machine', None)
+            tags['os'] = safe_get(post_data, 'os', None)
+            tags['platform'] = first(safe_get(sys_info, 'platform'
```

Comment:
You know, it is actually very mysterious why one would write a one-liner with a default value, and then use this default value explicitly as an argument? :thinking: 

Also, I can't understand why `pylint` did not catch those... There is a check for "using the default values in arguments".
Having to parse/count through nested braces is already a nasty exercise(hello, Lisp!). Finding a *default* values after all that effort is... disappointing.

Contrast 
```python

            tags['platform.details'] = first(safe_get(sys_info, 'platform.details',
                                                      None), None)
```
to 
```python

            tags['platform.details'] = first(safe_get(sys_info, 'platform.details'))
```

Which one is easier to understand?

Reply:
Yes, thank you to pointing this out. 
Changed.

>You know, it is actually very mysterious why one would write a one-liner with a default value, and then use this default value explicitly as an argument? 🤔

Because defaults in functions can be changed. 
My intention was to keep the behavior of this particular function unchanged and independent of function defaults.